### DISCUSSION TITLE
LVD editing

### DISCUSSION THREAD

**Discussion by CSharpM7** [ScanMountGoat/ssbh_editor#84] [Ideas]
I was curious if there are any future plans to be able to edit lvd files akin to StudioSB. Honestly, I think StudioSB+yamlvd solves most of my issues, so this isn't imperative at all, but it'd be interesting to see an updated version of StudioSB's lvd functions in this program

**Comment by ScanMountGoat** [comment]
I'm not opposed to adding support for LVD, but there aren't any plans to work on it at this time. Adding LVD support would require a number of changes to work well. This applies to implementations in other programs besides SSBH Editor.

- The lvd-rs library is incomplete and won't open/rebuild all LVD files. Adding unk fields until the files rebuild would be fine as a temporary fix. 
- The collision shapes are 2D, which needs to be handled properly in 3D programs. This isn't hard to do but takes some work to get right. SSBH Editor already has code for converting pixel coordinates on screen to and from 3D world coordinates. 
- Editing 2D shapes tends to work best with an orthographic camera, which could be added to SSBH Editor relatively easily.
- Drawing the LVD in the viewport requires a 3D LVD drawing library built with wgpu. This is straightforward but a bit too technical for most people to tackle.
- A UI needs to be created for manually editing the LVD file data. This is much easier to do in SSBH Editor than 3D content creation programs like Maya or Blender.
- There should be documentation on the fields in the LVD format itself and additional guides for how to use the editing functionality. This is also helpful for development of an LVD editor since I don't have much experience with the format.
- Decide which parts of the LVD format should be edited in the viewport and which can just use regular UI components like text boxes. 

I can pretty easily put together the rendering code, but the remaining points will require help from the community.

**Comment by ScanMountGoat** [comment]
This also doesn't need to be implemented all at once. One possible approach is to do this in stages.
1. Look into improving lvd-rs
2. Create a proof of concept editor using egui + lvd-rs as a standalone repo or a fork of ssbh_editor.
3. Incorporate the LVD editor into ssbh_editor.
4. Add rendering support for LVD to ssbh_editor.
5. Add viewport editing for drawing custom shapes.
